custom/src/main/java/com/solarwinds/opentelemetry/extensions/HostIdResourceProvider.java

```diff
@@ -0,0 +1,150 @@
+      setIfNotNull(builder::put, ResourceAttributes.K8S_POD_NAME, k8sMetadata.getPodName());
+    }
+
+    HostId.AwsMetadata awsMetadata = hostId.getAwsMetadata();
+    if (awsMetadata != null) {
+      setIfNotNull(builder::put, ResourceAttributes.HOST_ID, awsMetadata.getHostId());
+      setIfNotNull(builder::put, ResourceAttributes.HOST_NAME, awsMetadata.getHostName());
+      setIfNotNull(builder::put, ResourceAttributes.CLOUD_PROVIDER, awsMetadata.getCloudProvider());
+
+      setIfNotNull(
+          builder::put, ResourceAttributes.CLOUD_ACCOUNT_ID, awsMetadata.getCloudAccountId());
+      setIfNotNull(builder::put, ResourceAttributes.CLOUD_PLATFORM, awsMetadata.getCloudPlatform());
+      setIfNotNull(
+          builder::put,
+          ResourceAttributes.CLOUD_AVAILABILITY_ZONE,
+          awsMetadata.getCloudAvailabilityZone());
+
+      setIfNotNull(builder::put, ResourceAttributes.CLOUD_REGION, awsMetadata.getCloudRegion());
+      setIfNotNull(builder::put, ResourceAttributes.HOST_IMAGE_ID, awsMetadata.getHostImageId());
+      setIfNotNull(builder::put, ResourceAttributes.HOST_TYPE, awsMetadata.getHostType());
+    }
+
+    HostId.AzureVmMetadata azureVmMetadata = hostId.getAzureVmMetadata();
+    if (azureVmMetadata != null) {
+      setIfNotNull(builder::put, ResourceAttributes.HOST_ID, azureVmMetadata.getHostId());
+      setIfNotNull(builder::put, ResourceAttributes.HOST_NAME, azureVmMetadata.getHostName());
+      setIfNotNull(
+          builder::put, ResourceAttributes.CLOUD_PROVIDER, azureVmMetadata.getCloudProvider());
+
+      setIfNotNull(
+          builder::put, ResourceAttributes.CLOUD_ACCOUNT_ID, azureVmMetadata.getCloudAccountId());
+      setIfNotNull(
+          builder::put, ResourceAttributes.CLOUD_PLATFORM, azureVmMetadata.getCloudPlatform());
+      setIfNotNull(builder::put, ResourceAttributes.CLOUD_REGION, azureVmMetadata.getCloudRegion());
+
+      setIfNotNull(
+          builder::put, AttributeKey.stringKey("azure.vm.name"), azureVmMetadata.getAzureVmName());
+      setIfNotNull(
+          builder::put, AttributeKey.stringKey("azure.vm.size"), azureVmMetadata.getAzureVmSize());
+      setIfNotNull(
+          builder::put,
+          AttributeKey.stringKey("azure.resourcegroup.name"),
+          azureVmMetadata.getAzureResourceGroupName());
+
+      setIfNotNull(
+          builder::put,
+          AttributeKey.stringKey("azure.vm.scaleset.name"),
+          azureVmMetadata.getAzureVmScaleSetName());
+    }
+
+    setIfNotNull(builder::put, ResourceAttributes.HOST_NAME, hostId.getHostname());
+    setIfNotNull(
+        builder::put, ResourceAttributes.CLOUD_AVAILABILITY_ZONE, hostId.getEc2AvailabilityZone());
+    setIfNotNull(builder::put, ResourceAttributes.HOST_ID, hostId.getEc2InstanceId());
+    return Resource.create(builder.build());
+  }
+
+  private <V> void setIfNotNull(
+      BiConsumer<AttributeKey<V>, V> setter, AttributeKey<V> key, V value) {
```

Comment:
Another minor nitpick and more of a personal preference but given the setter is always the same in practice this feels a little more generic than it needs to be